### PR TITLE
Add all the unicom work I've done so far in PACE

### PR DESCRIPTION
**Pull request by deatheragetr**
Sharing my ostensibly working Unicom training loop code.  No guarantees of "correctness" in its current form.  However, it runs and the loss function optimizes over time.

Complete instructions are something like this:
1) Port this over to PACE to use cuda (relative directory structures should be maintained)
2) Download VLM4Bio dataset and run captioning notebook: https://github.com/BioCosmos-AI/BioCosmos/blob/c0201e4421be034108e36ede81a2432d27ce1b57/jupyter-workpad/thomas_scratchpad/Image-Captioning/image_caption_with_internvl.ipynb 
3) Run clustering and training notebook: https://github.com/BioCosmos-AI/BioCosmos/blob/c0201e4421be034108e36ede81a2432d27ce1b57/jupyter-workpad/thomas_scratchpad/Unicom-training/cluster_and_unicom_training.ipynb
4) Done?

### CODE REVIEW

File: jupyter-workpad/thomas_scratchpad/Unicom-training/retrieval.py [hunk 1]
```diff
@@ -0,0 +1,1056 @@
...
+    args = config
+    
+    # Initialize distributed processing with a single process
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = '12355'
+    distributed.init_process_group(backend='nccl', world_size=1, rank=0)
+
+    
+    # Load model and initialize
+    model, transform_clip = clip.load(args.model_name)
```
Comment (deatheragetr) [inline jupyter-workpad/thomas_scratchpad/Unicom-training/retrieval.py]: Do I need to use the original unicom-provided models?  This line requires some additional investigation from me.

Comment (deatheragetr) [reply jupyter-workpad/thomas_scratchpad/Unicom-training/retrieval.py]: Current conclusion is to use a pretrained base model, e.g., `ViT/B-32`.

**Comment by deatheragetr** [comment]
We're moving over to HiPerGator now that we've "proved" the efficacy of UNICOM + VLM4IO.  Merging because likely new code will be done elsewhere.
